featuretools/feature_base/feature_base.py

```diff
@@ -714,7 +714,10 @@ def copy(self):
     def generate_name(self):
         # exclude the groupby feature from base_names since it has a special
         # place in the feature name
-        base_names = [bf.get_name() for bf in self.base_features[:-1]]
+        if self.primitive.name == "cum_count":
+            base_names = [self.entity_id]
```

Comment:
The `Count` primitive uses this name scheme but I think there are complictions when this format is used with transform primitives. `Count` uses `COUNT(<path>)` instead of `COUNT(<entity.id>)`, and each path has a specific id variable associated with it.  

`CUM_COUNT(<entity.id>) by <groubpy_id>` could create multiple features with the same name if an entity has multiple id variables:
`CUM_COUNT(id_1.entity.id) by id_1` and `CUM_COUNT(id_2.entity.id) by id_1` will have identical feature names